stage0/src/CMakeLists.txt

```diff
@@ -282,6 +307,15 @@ endif()
 string(APPEND LEANC_STATIC_LINKER_FLAGS " ${LEAN_CXX_STDLIB}")
 string(APPEND LEANSHARED_LINKER_FLAGS " ${LEAN_CXX_STDLIB}")
 
+if(LLVM)
+    string(APPEND LEANSHARED_LINKER_FLAGS " -L${LLVM_CONFIG_LIBDIR} ${LLVM_CONFIG_LDFLAGS} ${LLVM_CONFIG_LIBS} ${LLVM_CONFIG_SYSTEM_LIBS}")
+    string(REPLACE "llvm-host" "llvm" LEANSHARED_LINKER_FLAGS ${LEANSHARED_LINKER_FLAGS})
```

Comment:
Note that this is incredibly janky, as we have the user pass us an `LLVM_CONFIG` path, after which we perform string replacement of `llvm-host` to `llvm`. We do this because:

- `llvm-config` prints paths relative to itself, and we're using the host's `llvm-config`, so `llvm-config` must live inside `llvm-host/`
- We need the paths to point to `llvm-target`, because that's where the correct version of the libs live.

I feel the right solution is to add two cmake flags, `-DLLVM_LINK_FLAGS`. `-DLLVM_CXX_FLAGS`. Then, in `script/prepare-llvm-*.sh`, we invoke `llvm-host/bin/llvm-config` to find the build flags, along with the required `sed` shenanigans to find-and-replace `llvm-host` with `llvm`

Reply:
@Kha : does the above strategy sound sensible to you?

Reply:
The current solution looks fine to me to be honest. Ultimately I feel it should be `llvm-config` itself that provides a solution for this, but it does not.

Reply:
@Kha The reason I am a tad uncomfortable with the current strategy is that on the off-chance someone downstream tries to build Lean4 with LLVM support, and they have `/path/to/llvm-config`  look like: `/path/to/llvm-host/path/to/llvm-config`, this path will be silently changed to `/path/to/llvm/path/to/llvm-config` (note the replacement of `llvm-host` -> `llvm`). I can imagine the resulting errors would be very confusing. 

I would suggest, as a happy middle ground, keeping the `string(REPLACE "llvm-host" "llvm" ...)` hack for this patch, and creating separate CMake flags `-DLLVM_{CXX,LINK}_FLAGS` in another patch?

Reply:
> they have `/path/to/llvm-config` look like: `/path/to/llvm-host/path/to/llvm-config`

I will gladly blame users for such a weird setup